docs/pages/identity-governance/idps/saml-guide.mdx

```diff
@@ -28,213 +28,231 @@ includes information about the application.
+![add service provider with entity descriptor](../../../img/access-controls/saml-idp/step-3-add-service-provider-ed.png)
 
-In the case of Simple Test Service Provider, which this guide is based on, the sample app is designed to grant access protected page
-for any well formatted IdP federated SAML assertion data.
+Click `Finish` button. The "IAMShowcase" app is now added to Teleport.
 
-As such, when you click **Finish** button in the previous step, the protected page of the Simple Test Service Provider is
-already available to access under resources page.
+<Admonition type="important">
+If an entity descriptor is provided, its content takes preference over values 
+provided for entity ID and ACS URL.
+</Admonition>
 
-## Step 3/3. Verify access to iamshowcase protected page
+## Step 4/4. Verify access to the protected page
 
-To verify everything works, navigate to **Resources** page in Teleport Web UI and look for the app.
+To verify everything works, navigate to the **Resources** page in the Teleport Web UI 
+and search for the "IAMShowcase" app.
 
-![Access app](../../../img/access-controls/saml-idp/access-app.png)
+![Access app](../../../img/access-controls/saml-idp/step-4-login.png)
 
-The "iamshowcase" app will now appear under resources tile. Inside this tile,
-click **Login** button, which will now forward you to iamshowcase protected page.
+From the "IAMShowcase" app tile, click the `Log in` button, which will forward 
+you to the IAMShowcase app's protected page with a SAML assertion data signed by 
+the Teleport SAML IdP.
 
-![Protected page access](../../../img/access-controls/saml-idp/protected-page-access.png)
+![Protected page access](../../../img/access-controls/saml-idp/step-4-service-provider-authenticated-page.png)
 
-This page shows Teleport user details along with other attributes such as roles are
-federated by Teleport SAML IdP.
+This page shows Teleport user details along with other attributes such as role 
+names that are federated by the Teleport SAML IdP, confirming a successful SAML 
+service provider configuration in Teleport.
 
-This demonstrates adding a service provider to Teleport configured with a working idp-initiated SSO login flow.
 
-## Recommended: Creating dedicated role to manage service provider
+## Manage SAML IdP service provider resource using `tctl`
 
-For production, we recommend creating a dedicated role to manage service provider.
+Examples of managing a `saml_idp_service_provider` resource using `tctl`.
 
-To create a dedicated role, first, ensure you are logged into Teleport as a user that has
-permissions to read and modify `saml_idp_service_provider` objects. The default `editor` role
-has access to this already, but in case you are using a more customized configuration,
-create a role called `sp-manager.yaml` with the following contents:
+### Create a `saml_idp_service_provider` resource
 
 ```yaml
-kind: role
+cat > iamshowcase.yaml << EOF
```

Comment:
Do we need the `cat` syntax in here or should we just use the raw YAML (which is what seems more common to me)?